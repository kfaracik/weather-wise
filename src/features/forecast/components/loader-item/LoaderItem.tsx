import {AnimatedViewBasic} from '@shared/components';
import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {ActivityIndicator} from 'react-native-paper';

export const LoaderItem = () => {
  return (
    <AnimatedViewBasic style={styles.loaderItem} disableExiting>
      <ActivityIndicator size="small" />
      <Text style={styles.text}>Loading...</Text>
    </AnimatedViewBasic>
  );
};

const styles = StyleSheet.create({
  loaderItem: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 8,
  },
  text: {
    marginLeft: 8,
    fontSize: 16,
  },
});
