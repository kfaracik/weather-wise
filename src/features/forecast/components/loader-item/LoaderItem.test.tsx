import React from 'react';
import {render} from '@testing-library/react-native';
import {LoaderItem} from './LoaderItem';

describe('LocationSearchScreen', () => {
  it('should generate snapshot', () => {
    const {toJSON} = render(<LoaderItem />);
    expect(toJSON()).toMatchSnapshot();
  });
});
