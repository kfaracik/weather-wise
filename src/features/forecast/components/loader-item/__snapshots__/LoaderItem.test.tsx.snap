// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationSearchScreen should generate snapshot 1`] = `
<View
  collapsable={false}
  entering={
    FadeIn {
      "build": [Function],
      "callbackV": undefined,
      "dampingRatioV": undefined,
      "dampingV": undefined,
      "delayV": undefined,
      "durationV": 200,
      "easingV": undefined,
      "initialValues": undefined,
      "massV": undefined,
      "overshootClampingV": undefined,
      "randomizeDelay": false,
      "reduceMotionV": "system",
      "restDisplacementThresholdV": undefined,
      "restSpeedThresholdV": undefined,
      "rotateV": undefined,
      "stiffnessV": undefined,
      "type": undefined,
    }
  }
  jestAnimatedStyle={
    {
      "value": {},
    }
  }
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 8,
    }
  }
>
  <View
    accessibilityRole="progressbar"
    accessibilityState={
      {
        "busy": true,
      }
    }
    accessible={true}
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "height": 24,
          "opacity": 1,
          "width": 24,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "height": 24,
              "transform": [
                {
                  "rotate": "45deg",
                },
              ],
              "width": 24,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "height": 12,
                "overflow": "hidden",
                "width": 24,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "height": 24,
                  "transform": [
                    {
                      "translateY": 0,
                    },
                    {
                      "rotate": "-165deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 12,
                    "overflow": "hidden",
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "borderColor": "rgba(103, 80, 164, 1)",
                      "borderRadius": 12,
                      "borderWidth": 2.4,
                      "height": 24,
                      "width": 24,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "height": 24,
              "transform": [
                {
                  "rotate": "45deg",
                },
              ],
              "width": 24,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "height": 12,
                "overflow": "hidden",
                "top": 12,
                "width": 24,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "height": 24,
                  "transform": [
                    {
                      "translateY": -12,
                    },
                    {
                      "rotate": "345deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 12,
                    "overflow": "hidden",
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "borderColor": "rgba(103, 80, 164, 1)",
                      "borderRadius": 12,
                      "borderWidth": 2.4,
                      "height": 24,
                      "width": 24,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "fontSize": 16,
          "marginLeft": 8,
        },
      ]
    }
  >
    Loading...
  </Text>
</View>
`;
